### PR TITLE
Implementation of Cardpart progress bar view

### PR DESCRIPTION
**Pull request by badrinathvm**
image

### CODE REVIEW

File: CardParts/src/Classes/Card Parts/CardPartProgressBarView.swift [hunk 1]
```diff
@@ -0,0 +1,454 @@
...
+    private var valuesBarHeight: CGFloat = 0
+    
+    private var markerLabelViewHeight: CGFloat = 0
+    
+    fileprivate var currentVal: Double = 0
+    
+    
+    public init(barValues:[Double], barColors: [UIColor], marker: UIView? = nil , markerLabelTitle: String, currentValue: Double, showShowBarValues: Bool) {
+        
+        self.barColors = barColors
```
Comment (croossin) [inline CardParts/src/Classes/Card Parts/CardPartProgressBarView.swift]: Can `barValues` and `barColors` be of different sizes? If not, can we protect against that?

File: CardParts/src/Classes/Card Parts/CardPartProgressBarView.swift [hunk 2]
```diff
@@ -0,0 +1,454 @@
...
+    }
+    
+    required public init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    
+    fileprivate func addMarker(marker: UIView?) {
+        let color: UIColor = UIColor.black
+        if marker != nil {
```
Comment (croossin) [inline CardParts/src/Classes/Card Parts/CardPartProgressBarView.swift]: why not `if let` so then we don't need to force cast? I get that it won't crash - but for syntax reasons

File: CardParts/src/Classes/Card Parts/CardPartProgressBarView.swift [hunk 3]
```diff
@@ -0,0 +1,454 @@
...
+        
+        updateConstraints()
+    }
+    
+    required public init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    
+    fileprivate func addMarker(marker: UIView?) {
```
Comment (croossin) [inline CardParts/src/Classes/Card Parts/CardPartProgressBarView.swift]: Do we need to deal with removing old markers? What if there is already a marker there?

File: CardParts/src/Classes/Card Parts/CardPartProgressBarView.swift [hunk 4]
```diff
@@ -0,0 +1,454 @@
...
+                self.view.layoutIfNeeded()
+            })
+        } else {
+            self.markerView.center.x += change
+            self.view.layoutIfNeeded()
+        }
+    }
+}
+
+extension Reactive where Base: CardPartProgressBarView {
```
Comment (croossin) [inline CardParts/src/Classes/Card Parts/CardPartProgressBarView.swift]: Awesome! Thanks for keeping bindability in mind!

File: CardParts/src/Extensions/UIColor.swift [hunk 5]
```diff
@@ -46,4 +46,11 @@ extension UIColor {
     public static var limeGreen: UIColor {get{return UIColor.colorFromHex(0x53B700)}}
     public static var yellowSea: UIColor {get{return UIColor.colorFromHex(0xFFAD00)}}
     public static var superNova: UIColor {get{return UIColor.colorFromHex(0xFFCA00)}}
+    
+    //progressBar View band colors
+    public static var sunflowerYellow: UIColor { get { return UIColor.colorFromHex(0xFFDC00)}}
```
Comment (croossin) [inline CardParts/src/Extensions/UIColor.swift]: Do we need these colors in this file?  I actually would like to start removing this file from CardParts as it pertains a lot to Intuit applications; opposed to making it generic for all applications.

File: CardParts/src/Extensions/UIFont.swift [hunk 6]
```diff
@@ -43,4 +43,6 @@ extension UIFont {
     class func turboGenericLightFontWithSize(_ size: CGFloat) -> UIFont {
```
Comment (croossin) [inline CardParts/src/Extensions/UIFont.swift]: More places that I want to remove and make more generic.  Should we make it not `turboGenericLightFontWithSize ` but instead `lightFontWithSize`

File: CardParts/src/Classes/Card Parts/CardPartProgressBarView.swift [hunk 7]
```diff
@@ -0,0 +1,454 @@
...
+    private var markerHeight: CGFloat = 7
+    
+    private var valuesBarHeight: CGFloat = 0
+    
+    private var markerLabelViewHeight: CGFloat = 0
+    
+    fileprivate var currentVal: Double = 0
+    
+    
+    public init(barValues:[Double], barColors: [UIColor], marker: UIView? = nil , markerLabelTitle: String, currentValue: Double, showShowBarValues: Bool) {
```
Comment (croossin) [inline CardParts/src/Classes/Card Parts/CardPartProgressBarView.swift]: Do we want to make `markerLabelTitle` optional?

File: CardParts/src/Extensions/UIFont.swift [hunk 8]
```diff
@@ -43,4 +43,6 @@ extension UIFont {
...
         return UIFont.systemFont(ofSize: size, weight: UIFont.Weight.light)
     }
+
+    static var titleTextMedium : UIFont {get{return UIFont.systemFont(ofSize: CGFloat(FontSize.x_Large.rawValue), weight: UIFont.Weight.medium)}}
```
Comment (croossin) [inline CardParts/src/Extensions/UIFont.swift]: Why this? Why not use `turboGenericMediumFont`?

File: README.md [hunk 9]
```diff
@@ -932,6 +933,19 @@ Provides the capability to add confetti with various types ( diamonds, star, mix
 <img src="https://raw.githubusercontent.com/Intuit/CardParts/master/images/confetti.gif" width="300" alt="Confetti"/>
 </p>
 
+### `CardPartProgressBarView`
+
+Provides the capability to configure different colors and custom marker , it's position to indicate the progress based ont he value provided.
```
Comment (croossin) [inline README.md]: Spelling error.  `on the` and the comma has a space after `marker` ;)

File: CardParts/src/Extensions/UIFont.swift (hunk 8 above)
Comment (badrinathvm) [reply CardParts/src/Extensions/UIFont.swift]: turbo wont  become generic  noe ? so created a new one.

File: README.md (hunk 9 above)
Comment (badrinathvm) [reply README.md]: Fixed.

File: CardParts/src/Extensions/UIFont.swift (hunk 8 above)
Comment (badrinathvm) [reply CardParts/src/Extensions/UIFont.swift]: Adopted

File: CardParts/src/Extensions/UIFont.swift (hunk 6 above)
Comment (badrinathvm) [reply CardParts/src/Extensions/UIFont.swift]: yes we should make it to `lightFontWithSize`

File: CardParts/src/Extensions/UIColor.swift (hunk 5 above)
Comment (badrinathvm) [reply CardParts/src/Extensions/UIColor.swift]: I will remove and use generic ones may be  for time being.

File: CardParts/src/Classes/Card Parts/CardPartProgressBarView.swift (hunk 4 above)
Comment (badrinathvm) [reply CardParts/src/Classes/Card Parts/CardPartProgressBarView.swift]: Thank you.

File: CardParts/src/Classes/Card Parts/CardPartProgressBarView.swift (hunk 2 above)
Comment (badrinathvm) [reply CardParts/src/Classes/Card Parts/CardPartProgressBarView.swift]: done, good eye.

File: CardParts/src/Classes/Card Parts/CardPartProgressBarView.swift (hunk 3 above)
Comment (badrinathvm) [reply CardParts/src/Classes/Card Parts/CardPartProgressBarView.swift]: Not required , marker view is optional and  mentioned during it's initialization.

File: CardParts/src/Classes/Card Parts/CardPartProgressBarView.swift (hunk 1 above)
Comment (badrinathvm) [reply CardParts/src/Classes/Card Parts/CardPartProgressBarView.swift]: It's customizable based on the values u provide. `   let barValues: [Double] = [300, 600, 650, 700, 750, 850]`

File: CardParts/src/Extensions/UIFont.swift (hunk 6 above)
Comment (badrinathvm) [reply CardParts/src/Extensions/UIFont.swift]: Can we do it as part of next PR's ?

File: CardParts/src/Classes/Card Parts/CardPartProgressBarView.swift (hunk 7 above)
Comment (badrinathvm) [reply CardParts/src/Classes/Card Parts/CardPartProgressBarView.swift]: yes we can make it. Done .

File: CardParts/src/Classes/Card Parts/CardPartProgressBarView.swift (hunk 3 above)
Comment (croossin) [reply CardParts/src/Classes/Card Parts/CardPartProgressBarView.swift]: No what if they keep calling addMarker... what happens

**Pull request merged**